src/Components/Components/src/Routing/RouteEntry.cs

```diff
@@ -62,8 +85,34 @@ internal void Match(RouteContext context)
                 }
             }
 
-            context.Parameters = parameters;
-            context.Handler = Handler;
+            // We track the number of segments in the template that matched
+            // against this particular route then only select the route that
+            // matches the most number of segments on the route that was passed.
+            // This check is an exactness check that favors the more precise of
+            // two templates in the event that the following route table exists.
+            //  Route 1: /{anythingGoes}
+            //  Route 2: /users/{id:int}
+            // And the provided route is `/users/1`. We want to choose Route 2
```

Comment:
The RouteTable exits as soon as it finds a match: https://github.com/dotnet/aspnetcore/blob/master/src/Components/Components/src/Routing/RouteTable.cs#L19-L23. Maybe we need to account for optionals when building the route table.

Reply:
Yes, seeing your other feedback some of the logic that is currently within the `Match` method can be removed if the comparator on the RouteTable is updated.